libs/hdf-converters/src/msft-secure-score-mapper.ts

```diff
@@ -0,0 +1,358 @@
+import {
+  SecureScore,
+  ControlScore,
+  SecureScoreControlProfile
+} from '@microsoft/microsoft-graph-types';
+import {ExecJSON} from 'inspecjs';
+import {version as HeimdallToolsVersion} from '../package.json';
+import {BaseConverter, ILookupPath, MappedTransform} from './base-converter';
+import * as _ from 'lodash';
+import {conditionallyProvideAttribute} from './utils/global';
+
+export type ProfileResponse = {
+  '@odata.context': string;
+  '@odata.nextLink': string;
+  value: SecureScoreControlProfile[];
+};
+
+export type SecureScoreResponse = {
+  '@odata.context': string;
+  '@odata.nextLink': string;
+  value: SecureScore[];
+};
+
+export type CombinedResponse = {
+  secureScore: SecureScoreResponse;
+  profiles: ProfileResponse;
+};
+
+export class MsftSecureScoreResults {
+  data: CombinedResponse;
+  withRaw: boolean;
+
+  constructor(combinedJson: string, withRaw = false) {
+    this.data = JSON.parse(combinedJson);
+    this.withRaw = withRaw;
+  }
+
+  toHdf(): ExecJSON.Execution[] {
+    const results: ExecJSON.Execution[] = [];
+
+    this.data.secureScore.value.forEach((element) => {
```

Comment:
prefer for..of in terms of loops, but here you could probably just chain maps

return this.data.securescore.value.map(e => whatever to get the new mapper).map(m => m.toHdf())

or even just

return this.data.securescore.value.map(e => get new mapper and then immediately call toHdf and return that)